From 90201c2bde4b0ca83cb07b7cc500885444ecef05 Mon Sep 17 00:00:00 2001
From: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue, 27 Feb 2024 10:19:45 -0300
Subject: [PATCH] prefcnt: cus__load_files() requires a non NULL conf_load

Since ff7bd7083f36db80 ("core: Allow sizing the loader hash table")
dwarf__load_files(), called by cus__load_files() needs a non-NULL
conf_load argument, add one to avoid a core dump.

J B reported this for syscse, which led me to do a quick audit to find
this other case.

Reported-by: J B <jb.1234abcd@gmail.com>
Link: https://lore.kernel.org/dwarves/CAJXMFhFbv5MccigKvmQt9MhYwKv2iyigEG3Bhs64BZ=PMWc0WQ@mail.gmail.com
Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
(cherry picked from commit 90201c2bde4b0ca83cb07b7cc500885444ecef05)
Upstream: https://git.kernel.org/pub/scm/devel/pahole/pahole.git/commit/?id=90201c2bde4b0ca83cb07b7cc500885444ecef05
---
 prefcnt.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/prefcnt.c b/prefcnt.c
index 8010afda..37ca18fd 100644
--- a/prefcnt.c
+++ b/prefcnt.c
@@ -14,6 +14,12 @@
 #include "dwarves.h"
 #include "dutil.h"
 
+static struct conf_fprintf conf;
+
+static struct conf_load conf_load = {
+	.conf_fprintf = &conf,
+};
+
 static void refcnt_tag(struct tag *tag, const struct cu *cu);
 
 static void refcnt_member(struct class_member *member, const struct cu *cu)
@@ -143,7 +149,7 @@ int main(int argc __maybe_unused, char *argv[])
 
 	dwarves__resolve_cacheline_size(NULL, 0);
 
-	err = cus__load_files(cus, NULL, argv + 1);
+	err = cus__load_files(cus, &conf_load, argv + 1);
 	if (err != 0) {
 		cus__fprintf_load_files_err(cus, "prefcnt", argv + 1, err, stderr);
 		return EXIT_FAILURE;
